RecoParticleFlow/PFProducer/interface/TableDefinitions.h

```diff
@@ -0,0 +1,191 @@
+#define RecoParticleFlow_PFProducer_TableDefinitions_h
+#include <set>
+#include <vector>
+
+#include "FWCore/SOA/interface/Table.h"
+#include "FWCore/SOA/interface/TableView.h"
+#include "FWCore/SOA/interface/Column.h"
+
+#include "DataFormats/Provenance/interface/ElementID.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
+
+template <class C>
+edm::ElementID refToElementID(const edm::Ref<C>& ref) {
+  return edm::ElementID(ref.id(), ref.index());
+}
+
+template <class C>
+edm::ElementID refToElementID(const edm::RefToBase<C>& ref) {
+  return edm::ElementID(ref.id(), ref.key());
+}
+
+namespace edm::soa {
+  namespace col::pf {
+    namespace track {
+      SOA_DECLARE_COLUMN(Pt, double, "Pt");
+      SOA_DECLARE_COLUMN(ExtrapolationValid, bool, "ExtrapolationValid");
+      SOA_DECLARE_COLUMN(Eta, double, "Eta");
+      SOA_DECLARE_COLUMN(Phi, double, "Phi");
+      SOA_DECLARE_COLUMN(Posx, double, "Posx");
+      SOA_DECLARE_COLUMN(Posy, double, "Posy");
+      SOA_DECLARE_COLUMN(Posz, double, "Posz");
+      SOA_DECLARE_COLUMN(PosR, double, "PosR");
+      SOA_DECLARE_COLUMN(ConvBremRefKey, edm::ElementID, "ConvBremRefKey");
+      SOA_DECLARE_COLUMN(ConvBremRefBaseKey, edm::ElementID, "ConvBremRefBaseKey");
+      SOA_DECLARE_COLUMN(IsLinkedToDisplacedVertex, bool, "IsLinkedToDisplacedVertex");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFKey, edm::ElementID, "GsfTrackRefPFKey");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFIsNonNull, bool, "GsfTrackRefPFIsNonNull");
+      SOA_DECLARE_COLUMN(KfPFRecTrackRefKey, edm::ElementID, "KfPFRecTrackRefKey");
+      SOA_DECLARE_COLUMN(KfPFRecTrackRefIsNonNull, bool, "KfPFRecTrackRefIsNonNull");
+      SOA_DECLARE_COLUMN(KfTrackRefKey, edm::ElementID, "KfTrackRefKey");
+      SOA_DECLARE_COLUMN(KfTrackRefBaseKey, edm::ElementID, "KfTrackRefBaseKey");
+      SOA_DECLARE_COLUMN(KfTrackRefIsNonNull, bool, "KfTrackRefIsNonNull");
+      SOA_DECLARE_COLUMN(ConvRefKey, edm::ElementID, "ConvRefKey");
+      SOA_DECLARE_COLUMN(ConvRefIsNonNull, bool, "ConvRefIsNonNull");
+      SOA_DECLARE_COLUMN(V0RefKey, edm::ElementID, "V0RefKey");
+      SOA_DECLARE_COLUMN(V0RefIsNonNull, bool, "V0RefIsNonNull");
+      SOA_DECLARE_COLUMN(TrackType_FROM_GAMMACONV, bool, "TrackType_FROM_GAMMACONV");
+      SOA_DECLARE_COLUMN(TrackType_FROM_V0, bool, "TrackType_FROM_V0");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFTrackId, int, "GsfTrackRefPFTrackId");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_TO_DISP_IsNonNull, bool, "DisplacedVertexRef_TO_DISP_IsNonNull");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_TO_DISP_Key, edm::ElementID, "DisplacedVertexRef_TO_DISP_Key");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_FROM_DISP_IsNonNull, bool, "DisplacedVertexRef_FROM_DISP_IsNonNull");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_FROM_DISP_Key, edm::ElementID, "DisplacedVertexRef_FROM_DISP_Key");
```

Comment:
what motivated this mix of the choice of style? (camel caps, underscores, all-caps)

Reply:
TO_DISP / FROM_DISP etc are enum values used elsewhere, I didn't want to introduce confusion by having also ToDisp = TO_DISP, but of course it's arguable what's better.